packages/dev-wallet/server/domains/authorization.js

```diff
@@ -0,0 +1,179 @@
+import {v4 as uuid} from "uuid"
+import * as db from "./user"
+import {signWithKey} from "../crypto"
+
+const PENDING = "PENDING"
+const APPROVED = "APPROVED"
+const DECLINED = "DECLINED"
+const AUTHORIZATIONS = {}
+
+const invariant = (fact, msg, ...rest) => {
+  if (!fact) {
+    const error = new Error(`INVARIANT ${msg}`)
+    error.stack = error.stack
+      .split("\n")
+      .filter(d => !/at invariant/.test(d))
+      .join("\n")
+    console.error("\n\n---\n\n", error, "\n\n", ...rest, "\n\n---\n\n")
+    throw error
+  }
+}
+
+export const createAuthorization = ({userId, transaction}) => {
+  invariant(userId, "createAuthorization({userId}) -- userId is required")
+  invariant(
+    transaction,
+    "createAuthorization({transaction}) -- transaction is required"
+  )
+
+  // Generate authorizationId, this will be used as a single lookup later
+  const authorizationId = uuid()
+
+  // Store authorization information
+  AUTHORIZATIONS[authorizationId] = {
+    authorizationId,
+    vsn: 0,
+    userId,
+    transaction,
+    status: PENDING,
+    reason: null,
+    compositeSignature: null,
+    // compositeSignature: {
```

Comment:
Should this be commented?

Reply:
yes, its there to remind me what needs to be in it for some upcoming stuff